src/qutip_cupy/cupy_dense.pyx

```diff
@@ -0,0 +1,42 @@
+#cython: language_level=3
+#cython: boundscheck=False, wraparound=False, initializedcheck=False
+
+
+cimport cython
+
+import numbers
+
+import cupy as cp 
+import qutip
+
+
+from qutip.core.data cimport base
+
+
+
+cdef class Dense(data.Data):
```

Comment:
You import only `base` from `qutip` so `data.Data` does not exist.

Reply:
Sorry about that. I have been trying many ways to import it. Still when changing to `base.Data` it is not finding the package, I changed the requirements as suggested by @jakelishman  on #7  but there seems to be still a linking problem.